### PR TITLE
[CIAPP] Document custom tags and metrics

### PR DESCRIPTION
**Pull request by cg14823**
### What does this PR do?
Documents how to use the custom tags and metrics feature in CI Visibility. It also add links for providers that can use this feature.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+## Adding tags to pipeline traces
+
+Tags can be added to the pipeline span or to the job span. To do this run:
+
+{{< code-block lang="bash" >}}
+datadog-ci tag [--level <pipeline|job>] [--tags <tags>]
+{{< /code-block >}}
+
+A valid [Datadog API key][2] has to be specified using the eniroment variable `DATADOG_API_KEY`.
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:44]:
```suggestion
A valid [Datadog API key][2] has to be specified using the environment variable `DATADOG_API_KEY`.
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+{{< site-region region="us5,us3,eu" >}}
+The [Datadog site][3] has to be specidied using the enviromental variable `DATADOG_SITE`.
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:47]:
```suggestion
The [Datadog site][3] has to be specified using the environment variable `DATADOG_SITE`.
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+## Adding metrics to pipeline traces
+
+Numerical tags can be added to the pipeline span or the job span. To do this run:
+
+{{< code-block lang="bash" >}}
+datadog-ci metric [--level <pipeline|job>] [--metrics <metrics>]
+{{< /code-block >}}
+
+A valid [Datadog API key][2] has to be specified using the eniroment variable `DATADOG_API_KEY`.
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:76]:
```suggestion
A valid [Datadog API key][2] has to be specified using the environment variable `DATADOG_API_KEY`.
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+{{< site-region region="us5,us3,eu" >}}
+The [Datadog site][3] has to be specidied using the enviromental variable `DATADOG_SITE`.
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:78]:
```suggestion
The [Datadog site][3] has to be specified using the enviroment variable `DATADOG_SITE`.
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+- GitLab (SaaS or self-hoster >= 14.1)
+- GitHub.com (SaaS). **Note:** For GitHub tags and metrics can only be added to the pipeline span
+
+## Installing the Datadog CI CLI
+
+Install the [`datadog-ci`][1] (>=v1.15.0) CLI globally using `npm`:
+
+{{< code-block lang="bash" >}}
+npm install -g @datadog/datadog-ci
+{{< /code-block >}}
```
[the remark this answers is not in the document]
Comment (cg14823) [reply content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:34]: I added a section with this info, good call.

**Comment by alai97** [comment]
Created DOCS-3536!

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    text: "Troubleshooting CI"
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">CI Visibility is not available for the selected site ({{< region-param key="dd_site_name" >}}).</div>
+{{< /site-region >}}
+
+<div class="alert alert-info">Custom tags and metrics is a beta feature and the API is still open to changes.</div>
+
+The custom tags and metrics commands provide a way to add user defined text and numerical tags to the CI Visibility
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:16]:
```suggestion
The custom tags and metrics commands provide a way to add user-defined text and numerical tags to your CI Visibility
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+pipeline traces.
+These tags can be used to create facets (string value tags) or measures (numerical value tags). Facets and measures
+can, in turn, be used to search, graph, or monitor the pipelines.
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:19]:
```suggestion
can then be used to search, graph, or monitor the pipelines.
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+## Compatibility
+
+Custom tags and metrics work with the following CI providers:
+
+- Buildkite
+- CircleCI
+- GitLab (SaaS or self-hoster >= 14.1)
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:27]:
```suggestion
- GitLab (SaaS or self-hosted >= 14.1)
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+- GitHub.com (SaaS). **Note:** For GitHub tags and metrics can only be added to the pipeline span
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:28]:
```suggestion
- GitHub.com (SaaS) **Note:** For GitHub, tags and metrics can only be added to the pipeline span.
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+## Installing the Datadog CI CLI
+
+Install the [`datadog-ci`][1] (>=v1.15.0) CLI globally using `npm`:
+
+{{< code-block lang="bash" >}}
+npm install -g @datadog/datadog-ci
+{{< /code-block >}}
+
+Alternatively you can try and use the beta [standalone binary][2] if you don't want to use `npm`
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:38]:
```suggestion
Alternatively, you can use the beta [standalone binary][2] if you don't want to use `npm`.
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+{{< code-block lang="bash" >}}
+Invoke-WebRequest -Uri "https://github.com/DataDog/datadog-ci/releases/latest/download/datadog-ci_win-x64.exe" -OutFile "datadog-ci.exe"
+{{< /code-block >}}
+{{% /tab %}}
+{{< /tabs >}}
+
+## Adding tags to pipeline traces
+
+Tags can be added to the pipeline span or to the job span. To do this run:
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:68]:
```suggestion
Tags can be added to the pipeline span or to the job span. To do this, run:
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+
+{{< code-block lang="bash" >}}
+datadog-ci tag [--level <pipeline|job>] [--tags <tags>]
+{{< /code-block >}}
+
+A valid [Datadog API key][3] has to be specified using the environment variable `DATADOG_API_KEY`.
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:74]:
```suggestion
You must specify a valid [Datadog API key][3] using the environment variable `DATADOG_API_KEY`.
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+
+{{< site-region region="us5,us3,eu" >}}
+The [Datadog site][1] has to be specified using the environment variable `DATADOG_SITE`.
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:77]:
```suggestion
You must specify the [Datadog site][1] using the environment variable `DATADOG_SITE`.
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+
+[1]: /getting_started/site/
+{{< /site-region >}}
+
+The example below adds the tag `team` to the pipeline span.
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:82]:
```suggestion
The following example adds the tag `team` to the pipeline span:
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+
+{{< code-block lang="bash" >}}
+datadog-ci tag --level pipeline --tags team:backend
+{{< /code-block >}}
+
+The example below adds the tag `go.version` to the span for the current job.
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:88]:
```suggestion
The following example adds the tag `go.version` to the span for the current job:
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+
+{{< code-block lang="bash" >}}
+datadog-ci tag --level job --tags "go.version:`go version`"
+{{< /code-block >}}
+
+You can then create facets from the tags by click on the gear next to the tag name in the [pipeline executions page][4]
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:94]:
```suggestion
To create facets from the tags, click the gear icon next to the tag name in the [pipeline executions page][4],
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+and then clicking on the `create facet` option.
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:95]:
```suggestion
and then click the **create facet** option.
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+
+{{< img src="ci/custom-tags-create-facet.mp4" alt="Facet creation for custom tag" style="width:100%;" video="true">}}
+
+## Adding metrics to pipeline traces
+
+Numerical tags can be added to the pipeline span or the job span. To do this run:
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:101]:
```suggestion
To add numerical tags to the pipeline span or the job span, run:
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+
+{{< code-block lang="bash" >}}
+datadog-ci metric [--level <pipeline|job>] [--metrics <metrics>]
+{{< /code-block >}}
+
+A valid [Datadog API key][3] has to be specified using the environment variable `DATADOG_API_KEY`.
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:107]:
```suggestion
You must specify a valid [Datadog API key][3] using the environment variable `DATADOG_API_KEY`.

```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+{{< site-region region="us5,us3,eu" >}}
+The [Datadog site][1] has to be specified using the environment variable `DATADOG_SITE`.
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:109]:
```suggestion
You must specify the [Datadog site][1] using the environment variable `DATADOG_SITE`.
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+
+[1]: /getting_started/site/
+{{< /site-region >}}
+
+The example below adds the metric `error_rate` to the pipeline span.
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:114]:
```suggestion
The following example adds the metric `error_rate` to the pipeline span:
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+
+{{< code-block lang="bash" >}}
+datadog-ci metric --level pipeline --metrics "error_rate:0.56"
+{{< /code-block >}}
+
+The example below adds a metric `binary.size` to the span for the currently running job.
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:120]:
```suggestion
The following example adds a metric `binary.size` to the span for the currently running job:
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 23]
```diff
@@ -0,0 +1,136 @@
...
+
+{{< code-block lang="bash" >}}
+datadog-ci metric --level job --metric "binary.size:`ls -l dst/binary | awk '{print \$5}' | tr -d '\n'`"
+{{< /code-block >}}
+
+You can create a measure by clicking on the gear next to the metrics name in the [pipeline executions page][4]
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:126]:
```suggestion
To create a measure, click the gear icon next to the metrics name in the [pipeline executions page][4]
```

File: content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md [hunk 24]
```diff
@@ -0,0 +1,136 @@
...
+and then clicking on the `create measure` option.
```
Comment (jtappa) [inline content/en/continuous_integration/setup_pipelines/custom_tags_and_metrics.md:127]:
```suggestion
and then click the **create measure** option.
```

**Review by jtappa — CHANGES_REQUESTED** [review]

**Review by kayayarai — APPROVED** [review]

**Pull request merged**